lib/routes/exposures/index.js

```diff
@@ -58,68 +128,231 @@ async function exposures(server, options, done) {
     handler: async request => {
       const { id } = request.authenticate()
       const { hash } = request.body
-      const control = hash.substr(0, 128)
-      const code = hash.substr(128)
 
-      const { rowCount: registerRateLimit } = await server.pg.write.query(
-        registerUpdate({
-          rateLimit: options.security.verifyRateLimit,
-          id
-        })
+      const controlHash = hash.substr(0, 128)
+      const codeHash = hash.substr(128)
+
+      const { token } = await exchangeCodeForToken(id, controlHash, codeHash)
+
+      return { token }
+    }
+  })
+
+  server.route({
+    method: 'POST',
+    url: '/verify',
+    schema: schema.exchange,
+    handler: async request => {
+      const { id } = request.authenticate()
+      const { code } = request.body
+
+      const controlHash = hash(code.substr(0, Math.floor(code.length / 2)))
+      const codeHash = hash(code)
+
+      const { onsetDate, testType, token } = await exchangeCodeForToken(
+        id,
+        controlHash,
+        codeHash
       )
 
-      if (registerRateLimit === 0) {
-        throw new TooManyRequests()
+      if (!onsetDate) {
+        throw new Forbidden()
```

Comment:
Migration `002.do.sql` drops the `NOT NULL` constraint, so this will be a necessary check. Null values are accepted by the old `POST /notify/positive` endpoint and it's important we don't change this for compatibility reasons.

Reply:
Ok. Are you making a subsequent change to https://github.com/covidgreen/covid-green-push-api/blob/current/lib/routes/notify/index.js#L75 to account for this? Or where are we going to handle checking for nulls?

Reply:
Yes - you can see the supporting change here https://github.com/covidgreen/covid-green-push-api/pull/10/files#diff-6021b95739a34ef9ef7e6ea09e57f02eR83